File: mobile-backend/src/main/java/ma/ac/emi/declare/jedeclaremobile/models/declarant/Adresse.java
package ma.ac.emi.declare.jedeclaremobile.models.declarant;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotBlank;

@Entity
@Data @NoArgsConstructor @AllArgsConstructor
public class Adresse {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String rue;

    @NotBlank
    private String ville;

    @NotBlank
    private String etat;

    @NotBlank
    @Length(min = 5)
    private String codePostal;
}

File: Web-FrontEnd/src/app/model/declaration/adresse.ts
export class Adresse {
  id: number;
  rue: string;
  ville: string;
  etat: string;
  codePostal: string;

  constructor(
    id: number,
    rue: string,
    ville: string,
    etat: string,
    codePostal: string
  ) {
    this.id = id;
    this.rue = rue;
    this.ville = ville;
    this.etat = etat;
    this.codePostal = codePostal;
  }
}
